test/format/smoke.sh

```diff
@@ -2,10 +2,15 @@
 
 set -e
 
+cache_max=$(free -m | grep -oP '\d+' | head -n 1)
```

Comment:
Not sure about the `-P` flag here - that's for a Perl regexp, which I'm not sure will work on BSDs (including macOS). Could we use the `[:digit:]` syntax and lose the `-P`?

Also - wouldn't `/proc/meminfo` be a better choice? Not sure about macOS though. I don't like that we rely on `free` printing outputs in a specific order.

Reply:
fair i have changed to using /proc/meminfo instead here. [03f0045](https://github.com/wiredtiger/wiredtiger/pull/10806/commits/03f004549589bb673ae066d52bf2916eb34f9630)

Reply:
Thanks!

Reply:
Discussed this IRL - format runs on macOS, the `format.sh` does not. So I'm happy with the current code.